### DISCUSSION TITLE
What us StreamingSoundData

### DISCUSSION THREAD

**Discussion by bq-wrongway** [tesselode/kira#88] [Q&A] [answered]
Hi,
I was wondering what streaming sound data is?
Like staticsounddata it has fn to read from file, i think that streaming sound data just uses some buffer or stream,while static one loads the whole file in memory?
Meaning if i want to play 5 sounds at same time,i would need to load 5 tracks to memory. 
Docs say this about streamingsounddata

**_A streaming sound that is not playing yet._**

Not super descriptive,i would really appreciate more info or tips about it,and when to use what.

Thank you for your work !

**Comment by tesselode** [comment, marked as answer]
The [sound module](https://docs.rs/kira/latest/kira/sound/index.html) has a bit more information, but I agree that there could be more discussion about when to use which type of sound.

`StaticSoundData` is more suited to short sounds, sounds you'll play multiple times, and sounds that need to be played with precise timing. `StreamingSoundData` is more suited to long sounds that you'll play one of, like background music.

**Reply by bq-wrongway** [reply]
Thanks, this does answer my question.

I did not manage to find any example for streaming data sound, im kinda struggling with it but i think thats out of scope. Thank you
